### DISCUSSION TITLE
Reference Zettel

### DISCUSSION THREAD

**Discussion by johanneskopton** [Zettlr/Zettlr#2937] [Ideas]
Hey lovely Zettlrers,

when I cite a source from my .bib library, I of course usually do that like this `@meadows1972`.

However, sometimes I have a dedicated zettel for a certain source (e.g. a book), where I store general thoughts about the book, that do not deserve their own Zettel.

If I cite the book now, I could also link this reference zettel like this `"The Limits to growth" [20210716142310]`, but this is kind of redundant (the title is in the reference zettel as well) and then I would not have the citation in the sidebar. Having both also seems redundant `@meadows1972 [20210716142310]`. This also would be inconsistent, if I created a reference zettel when there are already citations.

Currently, I think the only sensible way to handle this, would be the following (and as far as I know, this is how orgroam handles it):

- (somehow) detect reference zettel and map them to the .bib items they refer to
- have some link to the respective reference zettel at the citation, wherever there is one

To be as open for individual preferences as possible (like Zettlr thankfully always does its best to be), reference zettel detection could be via a configurable regex (just like the ID): whenever it finds the regex, the zettel automatically gets mapped to the first citation. So one could configure it do detect it by e. g.
```
---
reference: @meadows1972
---
# The Limits to Growth
```
or
```
# The Limits to Growth
#ref @meadows1972
```
or however one likes.

Okay, so now: Does anyone here even have the same problem? Are there other suggestions how to handle it?

Cheers,
Johannes

**Comment by nathanlesage** [comment]
Hasn't there been an update to the Club of Rome report recentlz …?

jk, to your topic: I was wondering this as well. Right now, I do it like such: First, I have a general reading note which I name according to the source, and also put a citekey to it in a references section down below (since I don't export it, I don't have to pay attention to how it might look afterwards). And then, whenever I need it, I can reference the reading note just by using the title, which is additionally more expressive than some ID.

Thus, I think it might not be necessary to couple a specific file to a citekey, and a certain amount of duplication is not at all harmful, I have found.

(After all, all I want is to reference my own thoughts across, not necessarily the source, as long as there is an easy way to find it, namely by following the link chain to the reading note)

**Reply by johanneskopton** [reply]
> Hasn't there been an update to the Club of Rome report recentlz …?

Maybe one or two :grinning: 

Ah, your approach makes sense to me. Since you start a general reading note as soon as you start reading something, you dont run into the problem of citing by citekey and therefore "missing out" a link to an existing reference zettel.

But I think, for me that would not work. I often cite or quote something, without reading or caring much about the entire opus. (Maybe that works better in engineering that in sociology?) Often, after finding lots of useful bits and pieces, I change my mind and read the whole damn thing, including the creation of general reading notes. Maybe its a really minor issue, but somehow it bothers me, that these reading notes are not connected to the previous citations.

I agree, that its far from a priority though :sweat_smile: But if I implemented it (and managed to do it properly), would there be a chance for it to get merged, or is it too special interest? :grin:

**Reply by nathanlesage** [reply]
Well, a reading note can also just contain fragments of the piece, so I don't see the problem :D 

I mean coupling citekeys and specific notes could lead into a hell of problems, so I'm not even sure it would be worth the effort.
